presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/optimization/function/OperatorTranslators.java

```diff
@@ -33,36 +34,60 @@ private OperatorTranslators()
+    @SupportedSignatures({
+            @SqlSignature(argumentType = StandardTypes.BIGINT, returnType = StandardTypes.BIGINT),
+            @SqlSignature(argumentType = StandardTypes.INTEGER, returnType = StandardTypes.INTEGER)})
+    public static JdbcExpression add(JdbcExpression left, JdbcExpression right)
     {
         return new JdbcExpression(infixOperation("+", left, right), forwardBindVariables(left, right));
     }
 
     @ScalarOperator(SUBTRACT)
-    @SqlType(StandardTypes.BIGINT)
-    public static JdbcExpression subtract(@SqlType(StandardTypes.BIGINT) JdbcExpression left, @SqlType(StandardTypes.BIGINT) JdbcExpression right)
+    @SupportedSignatures(@SqlSignature(argumentType = StandardTypes.BIGINT, returnType = StandardTypes.BIGINT))
+    public static JdbcExpression subtract(JdbcExpression left, JdbcExpression right)
     {
         return new JdbcExpression(infixOperation("-", left, right), forwardBindVariables(left, right));
     }
 
     @ScalarOperator(EQUAL)
-    @SqlType(StandardTypes.BOOLEAN)
-    public static JdbcExpression equal(@SqlType(StandardTypes.BIGINT) JdbcExpression left, @SqlType(StandardTypes.BIGINT) JdbcExpression right)
+    @SupportedSignatures({
+            @SqlSignature(argumentType = StandardTypes.BIGINT, returnType = StandardTypes.BOOLEAN),
+            @SqlSignature(argumentType = StandardTypes.INTEGER, returnType = StandardTypes.BOOLEAN),
+            @SqlSignature(argumentType = StandardTypes.SMALLINT, returnType = StandardTypes.BOOLEAN),
+            @SqlSignature(argumentType = StandardTypes.TINYINT, returnType = StandardTypes.BOOLEAN),
+            @SqlSignature(argumentType = StandardTypes.BOOLEAN, returnType = StandardTypes.BOOLEAN),
+            @SqlSignature(argumentType = StandardTypes.DATE, returnType = StandardTypes.BOOLEAN),
+            @SqlSignature(argumentType = StandardTypes.DECIMAL, returnType = StandardTypes.BOOLEAN),
+            @SqlSignature(argumentType = StandardTypes.REAL, returnType = StandardTypes.BOOLEAN),
+            @SqlSignature(argumentType = StandardTypes.DOUBLE, returnType = StandardTypes.BOOLEAN),
+            @SqlSignature(argumentType = StandardTypes.VARCHAR, returnType = StandardTypes.BOOLEAN),
+            @SqlSignature(argumentType = StandardTypes.TIMESTAMP, returnType = StandardTypes.BOOLEAN)})
+    public static JdbcExpression equal(JdbcExpression left, JdbcExpression right)
     {
         return new JdbcExpression(infixOperation("=", left, right), forwardBindVariables(left, right));
     }
 
     @ScalarOperator(NOT_EQUAL)
-    @SqlType(StandardTypes.BOOLEAN)
-    public static JdbcExpression notEqual(@SqlType(StandardTypes.BIGINT) JdbcExpression left, @SqlType(StandardTypes.BIGINT) JdbcExpression right)
+    @SupportedSignatures({
+            @SqlSignature(argumentType = StandardTypes.BIGINT, returnType = StandardTypes.BOOLEAN),
```

Comment:
Will this work if an input is a parametric type, like a list or struct?

Reply:
It should since we are using `parseTypeSignature`, which handles parametric types. I can work on tests for this as well